.github/workflows/build-images-and-create-deployment.yml

```diff
@@ -7,6 +7,10 @@ on:
       - staging
       - prod
 
+# add a concurrency group to prevent multiple builds from running at the same time
+concurrency:
+  group: $${{ github.ref }}-environment
+  cancel-in-progress: false
```

Comment:
curious, what is the issue with canceling "in progress" workflows?

Reply:
I don't know the exact reason but there is a lot of state changes that happens as a result of the deployment. At the very least there are test fixtures created and happy deployment configuration files that are modified. Interrupting a deployment could leave it in an unknown state. If we do cancel in progress workflows we should be mindful of when and where we allow it. I think determining that would be outside the scope of this PR.

Reply:
@atolopko-czi @Bento007 - `cancel-in-progress: true` seems like a blunt force instrument. It might open the [possibility of corrupting state](https://github.com/orgs/community/discussions/26311). I think it `cancel-in-progress: true` could work out for us, but I feel starting conservatively here is the best way to go.

Note that not setting `cancel-in-progress` (`false` is the default), won't prevent the `concurrency` feature from [canceling pending tasks](https://docs.github.com/en/actions/using-jobs/using-concurrency)

So we still save on time here by using the default `concurrency` feature

Reply:
Thanks, both. Good justifications. Maybe just add a TODO and/or issue to revisit this later?

Reply:
@atolopko-czi - Added a ticket here: https://github.com/chanzuckerberg/single-cell-data-portal/issues/5895